plugins/MakerbotWriter/MakerbotWriter.py

```diff
@@ -92,14 +104,18 @@ def write(self, stream: BufferedIOBase, nodes: List[SceneNode], mode=MeshWriter.
 
         gcode_text_io = StringIO()
         success = gcode_writer.write(gcode_text_io, None)
-
+        filename, filedata = "", ""
         # Writing the g-code failed. Then I can also not write the gzipped g-code.
         if not success:
             self.setInformation(gcode_writer.getInformation())
             return False
 
-        json_toolpaths = du.gcode_2_miracle_jtp(gcode_text_io.getvalue())
-        metadata = self._getMeta(nodes)
+        if file_format == "application/x-makerbot-sketch":
+            filename, filedata = "print.gcode", gcode_text_io.getvalue()
+            self._PNG_FORMATS = self._PNG_FORMAT
+        else:
+            filename, filedata = "print.jsontoolpath", du.gcode_2_miracle_jtp(gcode_text_io.getvalue())
+            self._PNG_FORMATS = self._PNG_FORMAT + self._PNG_FORMAT_METHOD
```

Comment:
```suggestion
        elif file_format == "application/x-makerbot-sketch":
            filename, filedata = "print.jsontoolpath", du.gcode_2_miracle_jtp(gcode_text_io.getvalue())
            self._PNG_FORMATS = self._PNG_FORMAT + self._PNG_FORMAT_METHOD
        else:
             raise Exception("Unsupported mime type") 
```
Or even better, use a [match](https://docs.python.org/3/tutorial/controlflow.html#match-statements) statement

Reply:
https://github.com/Ultimaker/Cura/pull/19029/commits/fdfc5dcc67f8efac60404421e86b357b5ef3a5af here 👯